packages/block-library/src/editor.scss

```diff
@@ -55,13 +55,13 @@
 
 :root .editor-styles-wrapper {
 	// Background colors.
-	@include background-colors();
```

Comment:
Are these mixins still necessary/used somewhere? Should we remove them as well?

Reply:
That's a development package, so I think you can just remove them and add a breaking change note to the changelog file

Reply:
The question I have is what happens when you instantiate a default block editor instance (just JS no backend), will you get decent defaults for theme.json configs/palettes/styles... How does that work?

Reply:
My understanding is that the `block-editor` was never concerned with providing the CSS for default colors, the CSS was there by other means.

Historically, that CSS was provided by the `block-library` and in current `trunk` is provided by the `block-library` and the server code for global styles. This PR removes the `block-library` CSS. It doesn't change anything for the `block-editor`. Is this your reading as well?

Reply:
Removed the unused mixins at https://github.com/WordPress/gutenberg/pull/33924/commits/ce60eae3ea92dca9f5b2fd84961a0bfc9bd2d57d Added a changelog note at https://github.com/WordPress/gutenberg/pull/33924/commits/dff2e85084b898ec734a1aad14b8565def68e0ee

Reply:
>  This PR removes the block-library CSS. It doesn't change anything for the block-editor. Is this your reading as well?

It seems like you're right, though it does bother me a bit because block-library is something third-party editors can easily load but the theme.json CSS is not.

Take a look at the playground for instance https://wordpress.github.io/gutenberg/?path=/story/playground-block-editor--default 

You can see that the block-library stylesheets are loaded there but with this PR they won't be. This won't break styles of the editor immediately because we add the "inline styles" for colors for backward compatibility but the classNames won't have any impact.

---
So it seems for me, this PR should be fine to land because it doesn't have a conceptual impact but it highlights one that we should tackle at some point. What should we do with the styles generated from theme.json: are these styles that the block editor should render based on its settings? Does the block editor work as it should without them? Should this style generation be baked or at least have a dedicated API/function in the frontend.

Reply:
Ok, I'd need a :+1: to land this then.

Do we want to add a breaking change to the `block-library` changelog as well? It's not directly an "API" but it seems consumers would be expecting those styles anyway.

I can look at what you mention as a follow-up. I'm around that area with https://github.com/WordPress/gutenberg/pull/32702 so can probably dig deeper.

Reply:
Created an issue for it at https://github.com/WordPress/gutenberg/issues/34005

Reply:
Yeah, let's add the changelog mention and ship.